File: scratch/search_keyword.py
import sys
import os

if len(sys.argv) < 3:
    print('Usage: python search_keyword.py <file> <query>')
    sys.exit(1)

file_name = sys.argv[1]
query = sys.argv[2].lower()

file_path = file_name if os.path.isabs(file_name) else os.path.join(os.path.dirname(__file__), '..', file_name)
if not os.path.exists(file_path):
    print(f'File does not exist: {file_path}')
    sys.exit(1)

with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
    lines = f.readlines()

count = 0
for index, line in enumerate(lines):
    if query in line.lower():
        print(f'{index + 1}: {line.strip()}')
        count += 1

print(f'\nFound {count} matches for "{query}" in {file_name}')

File: scratch/search_keyword.js
const fs = require('fs');
const path = require('path');

const file = process.argv[2];
const query = process.argv[3];

if (!file || !query) {
  console.log('Usage: node search_keyword.js <file> <query>');
  process.exit(1);
}

const filePath = path.isAbsolute(file) ? file : path.join(__dirname, '..', file);
if (!fs.existsSync(filePath)) {
  console.error(`File does not exist: ${filePath}`);
  process.exit(1);
}

const content = fs.readFileSync(filePath, 'utf8');
const lines = content.split('\n');

let count = 0;
lines.forEach((line, index) => {
  if (line.toLowerCase().includes(query.toLowerCase())) {
    console.log(`${index + 1}: ${line.trim()}`);
    count++;
  }
});
console.log(`\nFound ${count} matches for "${query}" in ${file}`);
